From b2921f1d0f59287a9609560d3b005b859ddc4af2 Mon Sep 17 00:00:00 2001
From: Roland Grunberg <rgrunber@redhat.com>
Date: Wed, 7 Sep 2016 15:21:25 -0400
Subject: [PATCH] Retrieve the underlying InputStream on newer versions of
 Jersey.

The underlying input stream, that supports writing seems to be nested
under an additional layer. This likely happened due to some upstream
change between Jersey 2.18 and 2.22.2.

Change-Id: I549061ec63054836f563a49b22b5df891015bf63
---
 .../eclipse/linuxtools/internal/docker/core/HttpHijackWorkaround.java   | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/containers/org.eclipse.linuxtools.docker.core/src/org/eclipse/linuxtools/internal/docker/core/HttpHijackWorkaround.java b/containers/org.eclipse.linuxtools.docker.core/src/org/eclipse/linuxtools/internal/docker/core/HttpHijackWorkaround.java
index ffe2169..441e7b4 100644
--- a/containers/org.eclipse.linuxtools.docker.core/src/org/eclipse/linuxtools/internal/docker/core/HttpHijackWorkaround.java
+++ b/containers/org.eclipse.linuxtools.docker.core/src/org/eclipse/linuxtools/internal/docker/core/HttpHijackWorkaround.java
@@ -46,6 +46,7 @@ public class HttpHijackWorkaround {
 				"input", //$NON-NLS-1$
 				"in", //$NON-NLS-1$
 				"in", //$NON-NLS-1$
+				"in", //$NON-NLS-1$
 				"wrappedStream", //$NON-NLS-1$
 				"in", //$NON-NLS-1$
 				"instream" //$NON-NLS-1$
@@ -57,6 +58,7 @@ public class HttpHijackWorkaround {
 				"org.glassfish.jersey.message.internal.EntityInputStream", //$NON-NLS-1$
 				FilterInputStream.class.getName(),
 				FilterInputStream.class.getName(),
+				FilterInputStream.class.getName(),
 				"org.apache.http.conn.EofSensorInputStream", //$NON-NLS-1$
 				"org.apache.http.impl.io.IdentityInputStream", //$NON-NLS-1$
 				"org.apache.http.impl.io.SessionInputBufferImpl" }; //$NON-NLS-1$
-- 
2.7.4

